rest-api/src/main/java/gov/cms/qpp/conversion/api/services/CpcFileServiceImpl.java

```diff
@@ -45,13 +47,31 @@
 	 */
 	public InputStreamResource getFileById(String fileId) throws IOException {
 		Metadata metadata = dbService.getMetadataById(fileId);
-		if (metadata != null && metadata.getCpc() && !metadata.getCpcProcessed()) {
+		if (isAnUnprocessedCpcFile(metadata)) {
 			return new InputStreamResource(storageService.getFileByLocationId(metadata.getSubmissionLocator()));
 		} else {
 			throw new NoFileInDatabaseException(FILE_NOT_FOUND);
 		}
 	}
 
+	/**
+	 * Process to ensure the file is an unprocessed cpc+ file and marks the file as processed
+	 *
+	 * @param fileId Identifier of the CPC+ file
+	 * @return Success or failure message.
+	 */
+	public String processFileById(String fileId) {
+		Metadata metadata = dbService.getMetadataById(fileId);
+		if (isAnUnprocessedCpcFile(metadata)) {
```

Comment:
I wonder if we should only throw the `NoFileInDatabaseException` if the `getMetadataById` call returns nothing.  If the file is not a CPC+ file, maybe a different exception.  I don't think we need to throw an exception at all if the file exists, is CPC+, but is already processed.  We can ignore that.

Reply:
Essentially all this is doing is enforcing a `File Not Found` message response back to the caller. The exception handler was a clean way of returning a new response type back with a message to the user.

Reply:
I think maybe passing a new message saying the file was already processed would be more beneficial.

Reply:
This has been changed to throw a `InvalidFileTypeException` for Non-cpc+ files. And return a "Already processed" type message for calls where a CPC+ file is already processed.